### PR TITLE
Updates to rekor verification

### PR DESCRIPTION
**Pull request by loosebazooka**
- Verify logId matches supplied public key for verification
- Clean up some error messages
- Use json canonicalizer instead of custom logic

Signed-off-by: Appu Goundan <appu@google.com>

### CODE REVIEW

**Comment by loosebazooka** [comment]
fixes #34

File: src/main/java/dev/sigstore/rekor/client/RekorVerifier.java [hunk 1]
```diff
@@ -58,25 +55,16 @@ public void verifyEntry(RekorEntry entry)
...
-    signableContent.put("integratedTime", entry.getIntegratedTime());
-    signableContent.put("logID", entry.getLogID());
-    signableContent.put("logIndex", entry.getLogIndex());
-
-    // TODO: I think we can verify the logID (sha256 of log public key) here too
-    // to provide the user with some useful information
-    // (https://github.com/sigstore/sigstore-java/issues/34)
-
-    var signableJson = GSON.get().toJson(signableContent);
+    String calculatedLogId = Hashing.sha256().hashBytes(rekorPublicKey.getEncoded()).toString();
```
Comment (patflynn) [inline src/main/java/dev/sigstore/rekor/client/RekorVerifier.java:58]: this could be a field maybe?

**Comment by patflynn** [comment]
I think this could become problematic once we have multiple tuf roots configurable, but we can tackle that once it's added. I'm still working out how it's supposed to work.

**Review by patflynn — APPROVED** [review]

**Comment by loosebazooka** [comment]
> I think this could become problematic once we have multiple tuf roots configurable, but we can tackle that once it's added. I'm still working out how it's supposed to work.

Yeah I think it's a bit hard to visualize this now. The other thing we need to think about is signing multiple artifacts -- should help with the maven work.

File: src/main/java/dev/sigstore/rekor/client/RekorVerifier.java [hunk 2]
```diff
@@ -44,6 +43,7 @@ public static RekorVerifier newRekorVerifier(byte[] rekorPublicKey)
   private RekorVerifier(PublicKey rekorPublicKey, String verifierAlgorithm) {
     this.rekorPublicKey = rekorPublicKey;
     this.verifierAlgorithm = verifierAlgorithm;
+    this.calculatedLogId = Hashing.sha256().hashBytes(rekorPublicKey.getEncoded()).toString();
```
Comment (patflynn) [inline src/main/java/dev/sigstore/rekor/client/RekorVerifier.java:46]: just FYI that Asra was telling me this probably won't work long term with changes coming down the pipe. I think it's fine for now.

**Review by patflynn — APPROVED** [review]

File: src/main/java/dev/sigstore/rekor/client/RekorVerifier.java (hunk 2 above)
Comment (loosebazooka) [reply src/main/java/dev/sigstore/rekor/client/RekorVerifier.java:46]: makes sense. we can deal with that as it happens. It's not an essential check either.

**Pull request merged**